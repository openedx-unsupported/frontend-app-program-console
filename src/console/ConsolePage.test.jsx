import { mount, shallow } from 'enzyme';
import { Collapsible } from '@edx/paragon';
import React from 'react';
import renderer from 'react-test-renderer';
import { ConsolePage } from './ConsolePage';
import ConnectedReportSection from '../report/reportSection';


describe('ConsolePage', () => {
  it('renders with the most basic props passed to it', () => {
    const consolePageComponent = (<ConsolePage
      authorized
      data={[]}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);
    const wrapper = mount(consolePageComponent);
    const tree = renderer.create(consolePageComponent);

    expect(tree).toMatchSnapshot();
    expect(wrapper.exists('.alert.alert-warning.show')).toEqual(false);
    expect(wrapper.exists('.alert.alert-danger.show')).toEqual(false);
    wrapper.unmount();
  });

  it('renders a warning banner if there is not an authorized user', () => {
    const consolePageComponent = (<ConsolePage
      authorized={false}
      data={[]}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);
    const wrapper = mount(consolePageComponent);
    const tree = renderer.create(consolePageComponent);

    expect(tree).toMatchSnapshot();
    expect(wrapper.exists('.alert.alert-warning.show')).toEqual(true);
    expect(wrapper.exists('.alert.alert-danger.show')).toEqual(false);
    wrapper.unmount();
  });

  it('renders an error banner when there was an error loading programs', () => {
    const consolePageComponent = (<ConsolePage
      authorized={false}
      loadingError="Request failed with HTTP 418"
      data={[]}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);
    const wrapper = mount(consolePageComponent);
    const tree = renderer.create(consolePageComponent);

    expect(tree).toMatchSnapshot();
    expect(wrapper.exists('.alert.alert-warning.show')).toEqual(false);
    expect(wrapper.exists('.alert.alert-danger.show')).toEqual(true);
    wrapper.unmount();
  });

  it('renders programs when there is data passed in', () => {
    const apiData = [{
      programKey: 'a',
      programTitle: 'a masters',
      programUrl: 'https://amasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }, {
      programKey: 'b',
      programTitle: 'b masters',
      programUrl: 'https://bmasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }];
    const consolePageComponent = (<ConsolePage
      authorized
      data={apiData}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);
    const wrapper = mount(consolePageComponent);
    const tree = renderer.create(consolePageComponent).toJSON();


    expect(tree).toMatchSnapshot();
    apiData.forEach((program, idx) => {
      expect(wrapper.find('h2').at(idx).text()).toEqual(program.programTitle);
      expect(wrapper.find(Collapsible).at(idx).prop('defaultOpen')).toEqual(true);
    });

    expect(wrapper.exists('.alert-danger.show')).toEqual(false);
    expect(wrapper.exists('.alert-info.show')).toEqual(false);
    expect(wrapper.exists('.alert-warning.show')).toEqual(false);

    wrapper.unmount();
  });

  it('passes isFirstSection=true to report section when there is no enrollment section', () => {
    const apiData = [{
      programKey: 'a',
      programTitle: 'a masters',
      programUrl: 'https://amasters.com',
      areEnrollmentsWritable: false,
      areReportsReadable: true,
    }];
    const consolePageComponent = (<ConsolePage
      authorized
      data={apiData}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);

    // shallow render ConsolePage to avoid fully rendering the ConnectedReportSection,
    // which requires a Redux store
    const wrapper = shallow(consolePageComponent);

    expect(wrapper.find('h2').text()).toEqual(apiData[0].programTitle);

    // we don't expect to see the enrollment section
    expect(wrapper.exists(Collapsible)).toEqual(false);

    expect(wrapper.find(ConnectedReportSection).prop('isFirstSection')).toEqual(true);


    expect(wrapper.exists('.alert-danger')).toEqual(false);
    expect(wrapper.exists('.alert-info')).toEqual(false);
    expect(wrapper.exists('.alert-warning.show')).toEqual(false);

    wrapper.unmount();
  });

  it('passes isFirstSection=false to report section when there is an enrollment section', () => {
    const apiData = [{
      programKey: 'a',
      programTitle: 'a masters',
      programUrl: 'https://amasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: true,
    }];
    const consolePageComponent = (<ConsolePage
      authorized
      data={apiData}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);

    // shallow render ConsolePage to avoid fully rendering the ConnectedReportSection,
    // which requires a Redux store
    const wrapper = shallow(consolePageComponent);

    expect(wrapper.find('h2').text()).toEqual(apiData[0].programTitle);

    const collapsible = wrapper.find(Collapsible);
    expect(collapsible.prop('className')).toEqual('shadow');
    expect(collapsible.prop('title')).toEqual('Manage Enrollments');
    expect(collapsible.prop('defaultOpen')).toEqual(true);

    expect(wrapper.find(ConnectedReportSection).prop('isFirstSection')).toEqual(false);

    expect(wrapper.exists('.alert-danger')).toEqual(false);
    expect(wrapper.exists('.alert-info')).toEqual(false);
    expect(wrapper.exists('.alert-warning.show')).toEqual(false);

    wrapper.unmount();
  });

  it('renders program banners when they are included', () => {
    const apiData = [{
      programKey: 'a',
      programTitle: 'a masters',
      programUrl: 'https://amasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }, {
      programKey: 'b',
      programTitle: 'b masters',
      programUrl: 'https://bmasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }];

    const programBanners = {
      a: [{
        id: `a${Date.now()}`,
        bannerType: 'danger',
        message: 'Sorry something went wrong',
      }],
      b: [{
        id: `b${Date.now()}`,
        bannerType: 'success',
        message: 'You did it!',
      }],
    };
    const consolePageComponent = (<ConsolePage
      authorized
      data={apiData}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={programBanners}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);
    const wrapper = mount(consolePageComponent);
    const tree = renderer.create(consolePageComponent).toJSON();

    expect(tree).toMatchSnapshot();
    expect(wrapper.find('.alert-danger.show .alert-dialog').at(0).text()).toEqual('Sorry something went wrong ');
    expect(wrapper.find('.alert-success.show .alert-dialog').at(0).text()).toEqual('You did it! ');

    wrapper.unmount();
  });

  it('calls the fetchPrograms function on pageload', () => {
    const mock = jest.fn();

    expect(mock).not.toHaveBeenCalled();

    mount(<ConsolePage
      authorized
      data={[]}
      downloadEnrollments={() => {}}
      fetchPrograms={mock}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);

    expect(mock).toHaveBeenCalled();
  });

  it('calls the correct action with the correct program key on download button clicks', () => {
    const mock = jest.fn();

    const apiData = [{
      programKey: 'a',
      programTitle: 'a masters',
      programUrl: 'https://amasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }, {
      programKey: 'b',
      programTitle: 'b masters',
      programUrl: 'https://bmasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }];

    const wrapper = mount(<ConsolePage
      authorized
      data={apiData}
      downloadEnrollments={mock}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={() => {}}
      removeBanner={() => {}}
    />);

    const programADownloadProgramButton = wrapper.find('button.btn.btn-primary').at(0);
    expect(programADownloadProgramButton.text()).toEqual('Download Program Enrollments');
    programADownloadProgramButton.simulate('click');
    expect(mock).toHaveBeenCalledWith('a', false);

    const programADownloadCourseButton = wrapper.find('button.btn.btn-primary').at(1);
    expect(programADownloadCourseButton.text()).toEqual('Download Course Enrollments');
    programADownloadCourseButton.simulate('click');
    expect(mock).toHaveBeenCalledWith('a', true);

    const programBDownloadProgramButton = wrapper.find('button.btn.btn-primary').at(2);
    expect(programBDownloadProgramButton.text()).toEqual('Download Program Enrollments');
    programBDownloadProgramButton.simulate('click');
    expect(mock).toHaveBeenCalledWith('b', false);

    const programBDownloadCourseButton = wrapper.find('button.btn.btn-primary').at(3);
    expect(programBDownloadCourseButton.text()).toEqual('Download Course Enrollments');
    programBDownloadCourseButton.simulate('click');
    expect(mock).toHaveBeenCalledWith('b', true);
  });

  it('calls the correct action with the correct program key onchange of the file inputs', () => {
    const mock = jest.fn();

    const apiData = [{
      programKey: 'a',
      programTitle: 'a masters',
      programUrl: 'https://amasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }, {
      programKey: 'b',
      programTitle: 'b masters',
      programUrl: 'https://bmasters.com',
      areEnrollmentsWritable: true,
      areReportsReadable: false,
    }];

    const wrapper = mount(<ConsolePage
      authorized
      data={apiData}
      downloadEnrollments={() => {}}
      fetchPrograms={() => {}}
      programBanners={{}}
      uploadEnrollments={mock}
      removeBanner={() => {}}
    />);

    const file = new File([], 'test');

    expect(mock).not.toHaveBeenCalled();
    wrapper.find('.input-overlay-hack').at(0).simulate('change', { target: { name: 'pollName', files: [file] } });
    expect(mock).toHaveBeenCalledWith('a', false, file);
    wrapper.find('.input-overlay-hack').at(1).simulate('change', { target: { name: 'pollName', files: [file] } });
    expect(mock).toHaveBeenCalledWith('a', true, file);
    wrapper.find('.input-overlay-hack').at(2).simulate('change', { target: { name: 'pollName', files: [file] } });
    expect(mock).toHaveBeenCalledWith('b', false, file);
    wrapper.find('.input-overlay-hack').at(3).simulate('change', { target: { name: 'pollName', files: [file] } });
    expect(mock).toHaveBeenCalledWith('b', true, file);
  });
});
