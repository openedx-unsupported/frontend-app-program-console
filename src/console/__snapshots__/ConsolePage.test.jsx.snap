// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsolePage renders a warning banner if there is not an authorized user 1`] = `
<div
  className="container half-width-element py-5 align-items-start"
>
  <h1>
    Program Console
  </h1>
  <div
    className="alert fade alert-danger"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <div>
        <p>
          An error was encountered while loading your program list: 
          <em>
            null
          </em>
        </p>
        <p>
          Please try waiting a moment and then refreshing the page. If the issue persists, please reach out to 
          <a
            href="mailto:partner-support@edx.org"
          >
            partner-support@edx.org
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    className="alert fade alert-warning show"
    hidden={false}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <p>
        It appears you do not have proper permissions to access this application. Please reach out to 
        <a
          href="mailto:partner-support@edx.org"
        >
          partner-support@edx.org
        </a>
         requesting access to the Registrar service.
      </p>
    </div>
  </div>
</div>
`;

exports[`ConsolePage renders an error banner when there was an error loading programs 1`] = `
<div
  className="container half-width-element py-5 align-items-start"
>
  <h1>
    Program Console
  </h1>
  <div
    className="alert fade alert-danger show"
    hidden={false}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <div>
        <p>
          An error was encountered while loading your program list: 
          <em>
            Request failed with HTTP 418
          </em>
        </p>
        <p>
          Please try waiting a moment and then refreshing the page. If the issue persists, please reach out to 
          <a
            href="mailto:partner-support@edx.org"
          >
            partner-support@edx.org
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    className="alert fade alert-warning"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <p>
        It appears you do not have proper permissions to access this application. Please reach out to 
        <a
          href="mailto:partner-support@edx.org"
        >
          partner-support@edx.org
        </a>
         requesting access to the Registrar service.
      </p>
    </div>
  </div>
</div>
`;

exports[`ConsolePage renders program banners when they are included 1`] = `
<div
  className="container half-width-element py-5 align-items-start"
>
  <h1>
    Program Console
  </h1>
  <div
    className="alert fade alert-danger"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <div>
        <p>
          An error was encountered while loading your program list: 
          <em>
            null
          </em>
        </p>
        <p>
          Please try waiting a moment and then refreshing the page. If the issue persists, please reach out to 
          <a
            href="mailto:partner-support@edx.org"
          >
            partner-support@edx.org
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    className="alert fade alert-warning"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <p>
        It appears you do not have proper permissions to access this application. Please reach out to 
        <a
          href="mailto:partner-support@edx.org"
        >
          partner-support@edx.org
        </a>
         requesting access to the Registrar service.
      </p>
    </div>
  </div>
  <div
    className="container mb-4"
  >
    <h2>
      a masters
    </h2>
    <div
      className="alert fade alert-dismissible alert-danger show"
      hidden={false}
      role="alert"
    >
      <button
        aria-label="Close"
        className="btn close"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      <div
        className="alert-dialog"
      >
        <div
          className="modal-alert"
        >
          Sorry something went wrong 
        </div>
      </div>
    </div>
    <div
      className="pgn_collapsible shadow collapsible-card is-open"
      open={undefined}
    >
      <div
        aria-expanded={true}
        className="collapsible-trigger"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span>
          Manage Enrollments
        </span>
        <span
          className="ml-2"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-minus fa-w-14 "
            data-icon="minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        >
          <div
            className="collapsible-body"
          >
            <div
              className="container"
            >
              <div
                className="row mb-3 mb-md-2 "
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Program Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Program Enrollments
                  </button>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Course Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Course Enrollments
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container mb-4"
  >
    <h2>
      b masters
    </h2>
    <div
      className="alert fade alert-dismissible alert-success show"
      hidden={false}
      role="alert"
    >
      <button
        aria-label="Close"
        className="btn close"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      <div
        className="alert-dialog"
      >
        <div
          className="modal-alert"
        >
          You did it! 
        </div>
      </div>
    </div>
    <div
      className="pgn_collapsible shadow collapsible-card is-open"
      open={undefined}
    >
      <div
        aria-expanded={true}
        className="collapsible-trigger"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span>
          Manage Enrollments
        </span>
        <span
          className="ml-2"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-minus fa-w-14 "
            data-icon="minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        >
          <div
            className="collapsible-body"
          >
            <div
              className="container"
            >
              <div
                className="row mb-3 mb-md-2 "
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Program Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Program Enrollments
                  </button>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Course Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Course Enrollments
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConsolePage renders programs when there is data passed in 1`] = `
<div
  className="container half-width-element py-5 align-items-start"
>
  <h1>
    Program Console
  </h1>
  <div
    className="alert fade alert-danger"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <div>
        <p>
          An error was encountered while loading your program list: 
          <em>
            null
          </em>
        </p>
        <p>
          Please try waiting a moment and then refreshing the page. If the issue persists, please reach out to 
          <a
            href="mailto:partner-support@edx.org"
          >
            partner-support@edx.org
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    className="alert fade alert-warning"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <p>
        It appears you do not have proper permissions to access this application. Please reach out to 
        <a
          href="mailto:partner-support@edx.org"
        >
          partner-support@edx.org
        </a>
         requesting access to the Registrar service.
      </p>
    </div>
  </div>
  <div
    className="container mb-4"
  >
    <h2>
      a masters
    </h2>
    <div
      className="pgn_collapsible shadow collapsible-card is-open"
      open={undefined}
    >
      <div
        aria-expanded={true}
        className="collapsible-trigger"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span>
          Manage Enrollments
        </span>
        <span
          className="ml-2"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-minus fa-w-14 "
            data-icon="minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        >
          <div
            className="collapsible-body"
          >
            <div
              className="container"
            >
              <div
                className="row mb-3 mb-md-2 "
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Program Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Program Enrollments
                  </button>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Course Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Course Enrollments
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container mb-4"
  >
    <h2>
      b masters
    </h2>
    <div
      className="pgn_collapsible shadow collapsible-card is-open"
      open={undefined}
    >
      <div
        aria-expanded={true}
        className="collapsible-trigger"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span>
          Manage Enrollments
        </span>
        <span
          className="ml-2"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-minus fa-w-14 "
            data-icon="minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        >
          <div
            className="collapsible-body"
          >
            <div
              className="container"
            >
              <div
                className="row mb-3 mb-md-2 "
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Program Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Program Enrollments
                  </button>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md"
                >
                  <div
                    className="btn btn-primary w-100 mb-1"
                  >
                    <input
                      className="sr input-overlay-hack"
                      onChange={[Function]}
                      type="file"
                    />
                    Upload Course Enrollments
                  </div>
                </div>
                <div
                  className="col-md"
                >
                  <button
                    className="btn btn-primary w-100"
                    onClick={[Function]}
                  >
                    Download Course Enrollments
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConsolePage renders with the most basic props passed to it 1`] = `
<div
  className="container half-width-element py-5 align-items-start"
>
  <h1>
    Program Console
  </h1>
  <div
    className="alert fade alert-danger"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <div>
        <p>
          An error was encountered while loading your program list: 
          <em>
            null
          </em>
        </p>
        <p>
          Please try waiting a moment and then refreshing the page. If the issue persists, please reach out to 
          <a
            href="mailto:partner-support@edx.org"
          >
            partner-support@edx.org
          </a>
          .
        </p>
      </div>
    </div>
  </div>
  <div
    className="alert fade alert-warning"
    hidden={true}
    role="alert"
  >
    <div
      className="alert-dialog"
    >
      <p>
        It appears you do not have proper permissions to access this application. Please reach out to 
        <a
          href="mailto:partner-support@edx.org"
        >
          partner-support@edx.org
        </a>
         requesting access to the Registrar service.
      </p>
    </div>
  </div>
</div>
`;
